src/dsa/rpo_falcon512/keys/mod.rs

```diff
@@ -0,0 +1,51 @@
+use super::{
+    math::{FalconFelt, Polynomial},
+    ByteReader, ByteWriter, Deserializable, DeserializationError, Felt, Serializable, Signature,
+    Word,
+};
+
+mod public_key;
+pub use public_key::{PubKeyPoly, PublicKey};
+
+mod secret_key;
+pub use secret_key::SecretKey;
+
+// TESTS
+// ================================================================================================
+
+#[cfg(all(test, feature = "std"))]
+mod tests {
+    use super::{Felt, SecretKey, Word};
+    use rand::rngs::OsRng;
+    use winter_utils::{Deserializable, Serializable};
+
+    #[test]
+    fn test_falcon_verification() {
+        // generate random keys
+        let sk = SecretKey::new();
+        let pk = sk.public_key();
```

Comment:
In the next PR, we should change this to run the test in `no-std` context as well. For this, we need to use `SecretKey::with_rng()` constructor. For the `rng` parameter, we can use `RpoRandomCoin` (instantiated with some dummy seed) - but we'll first need to implement `Rng` trait for it.